4.8/aspnet/windowsservercore-2004/Dockerfile

```diff
@@ -0,0 +1,31 @@
+# escape=`
+
+ARG REPO=mcr.microsoft.com/dotnet/framework/runtime
+FROM $REPO:4.8-windowsservercore-2004
+
+SHELL ["powershell", "-Command", "$ErrorActionPreference = 'Stop'; $ProgressPreference = 'SilentlyContinue';"]
+
+RUN Add-WindowsFeature Web-Server; `
+    Add-WindowsFeature NET-Framework-45-ASPNET; `
+    Add-WindowsFeature Web-Asp-Net45; `
+    Remove-Item -Recurse C:\inetpub\wwwroot\*; `
+    Invoke-WebRequest -Uri https://dotnetbinaries.blob.core.windows.net/servicemonitor/2.0.1.6/ServiceMonitor.exe -OutFile C:\ServiceMonitor.exe
+
+# Install Roslyn compilers and ngen binaries
+RUN Invoke-WebRequest https://api.nuget.org/packages/microsoft.net.compilers.2.9.0.nupkg -OutFile c:\microsoft.net.compilers.2.9.0.zip; `
+    Expand-Archive -Path c:\microsoft.net.compilers.2.9.0.zip -DestinationPath c:\RoslynCompilers; `
+    Remove-Item c:\microsoft.net.compilers.2.9.0.zip -Force; `
+    &C:\Windows\Microsoft.NET\Framework64\v4.0.30319\ngen.exe install c:\RoslynCompilers\tools\csc.exe /ExeConfig:c:\RoslynCompilers\tools\csc.exe | `
+    &C:\Windows\Microsoft.NET\Framework64\v4.0.30319\ngen.exe install c:\RoslynCompilers\tools\vbc.exe /ExeConfig:c:\RoslynCompilers\tools\vbc.exe | `
+    &C:\Windows\Microsoft.NET\Framework64\v4.0.30319\ngen.exe install c:\RoslynCompilers\tools\VBCSCompiler.exe /ExeConfig:c:\RoslynCompilers\tools\VBCSCompiler.exe | `
+    &C:\Windows\Microsoft.NET\Framework\v4.0.30319\ngen.exe install c:\RoslynCompilers\tools\csc.exe /ExeConfig:c:\RoslynCompilers\tools\csc.exe | `
```

Comment:
Should the 32-bit assemblies get ngen'd in the images when they aren't in the base image?

What is all ngen'd with `ngen update`?  IIRC this was done because something in `Add-WindowsFeature` was queuing up some assemblies to be ngen'd.  I think this should be investigated to see if it is necessary.

This layer is 138 MB (expanded) in the 4.8-1909 images.  It would be good to get a check on the size.

Reply:
There are some 32-bit assemblies that are ngen'd in the Windows base image: `mscorlib`, `System`, and `System.Core` (the same set exists for 64-bit).  Executing `Add-WindowsFeature Web-Server` results in those assemblies needing to be ngen'd again.  So I'm actually going to move the `ngen update` calls up to that layer.  It's not going to be necessary to call `ngen update` for the Roslyn layer.

After removing the explicit ngen of 32-bit Roslyn assemblies and the `ngen update` calls, the Roslyn/ngen layer comes out to 92 MB.

I'll include an overall layer size analysis as a separate comment in the PR.